Sail/src/Sail/entities/Entity.cpp

```diff
@@ -1,14 +1,26 @@
 #include "pch.h"
 #include "Entity.h"
 
-Entity::SPtr Entity::Create() {
-	return std::make_shared<Entity>();
+Entity::SPtr Entity::Create(const std::string& name) {
+	return std::make_shared<Entity>(name);
 }
 
 Entity::Entity() {
+	m_name = "";
+}
 
+Entity::Entity(const std::string& name) {
```

Comment:
```suggestion
Entity::Entity(const std::string& name) 
	: m_name(name)
{
```
Moved variable initialization to initializer list